packages/phone-number-privacy/combiner/test/unit/domainThresholdState.test.ts

```diff
@@ -0,0 +1,148 @@
+
+describe('domain threshold state', () => {
+  const expectedVersion = getVersion()
+  const domainThresholdStateService = new DomainThresholdStateService(config.domains)
+
+  // TODO: add tests with failed signer responses, depending on
+  // result of https://github.com/celo-org/celo-monorepo/issues/9826
+
+  const getSession = (domainStates: SequentialDelayDomainState[]) => {
+    const mockRequest = {
+      body: {},
+    } as Request
+    const mockResponse = {
+      locals: {
+        logger: new Logger({ name: 'logger' }),
+      },
+    } as Response
+    const session = new Session(mockRequest, mockResponse)
+    domainStates.forEach((status) => {
+      const res: DomainRestrictedSignatureResponseSuccess | DomainQuotaStatusResponseSuccess = {
+        success: true,
+        version: expectedVersion,
+        status,
+      }
+      session.responses.push({ url: 'random url', res, status: 200 })
+    })
+    return session
+  }
+
+  const now = Date.now()
+  const timer = now - 1
+  const counter = 2
+
+  const varyingDomainStates = [
+    {
+      statuses: [
+        { timer, counter: 2, disabled: false, now },
+        { timer, counter: 2, disabled: false, now },
+        { timer, counter: 2, disabled: false, now },
+      ],
+      expectedCounter: 2,
+      expectedTimer: timer,
+    },
+    {
+      statuses: [
+        { timer, counter: 1, disabled: false, now },
+        { timer, counter: 2, disabled: false, now },
+        { timer, counter: 2, disabled: false, now },
+      ],
+      expectedCounter: 2,
+      expectedTimer: timer,
+    },
+    {
+      statuses: [
+        { timer, counter: 0, disabled: false, now },
+        { timer, counter: 1, disabled: false, now },
+        { timer, counter: 2, disabled: false, now },
+      ],
+      expectedCounter: 1,
+      expectedTimer: timer,
+    },
+    {
+      statuses: [
+        { timer, counter: 0, disabled: true, now },
+        { timer, counter: 1, disabled: false, now },
+        { timer, counter: 2, disabled: false, now },
+      ],
+      expectedCounter: 2,
+      expectedTimer: timer,
+    },
+    {
+      statuses: [
+        { timer: timer - 1, counter, disabled: false, now },
+        { timer, counter, disabled: false, now },
+        { timer, counter, disabled: false, now },
+      ],
+      expectedCounter: counter,
+      expectedTimer: timer,
+    },
+    {
+      statuses: [
+        { timer: timer - 1, counter, disabled: false, now },
+        { timer: timer - 1, counter, disabled: false, now },
+        { timer, counter, disabled: false, now },
+      ],
+      expectedCounter: counter,
+      expectedTimer: timer - 1,
+    },
+    {
+      statuses: [
+        { timer: timer - 1, counter: 1, disabled: false, now },
+        { timer, counter: 1, disabled: false, now },
+        { timer, counter: 2, disabled: false, now },
+      ],
+      expectedCounter: 1,
+      expectedTimer: timer,
```

Comment:
similar to above comment, the current code returns `timer`, but shouldn't this scenario return `timer-1`?